src/main/java/nl/knaw/dans/filemigration/db/EasyFileDAO.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2021 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.filemigration.db;
+
+import io.dropwizard.hibernate.AbstractDAO;
+import nl.knaw.dans.filemigration.api.EasyFile;
+import org.hibernate.SessionFactory;
+
+import java.util.List;
+
+public class EasyFileDAO extends AbstractDAO<EasyFile> {
+
+  public EasyFileDAO(SessionFactory sessionFactory) {
+    super(sessionFactory);
+  }
+
+  public List<EasyFile> findByDatasetId(String id) {
+    return list(namedTypedQuery(EasyFile.FIND_BY_DATASET_ID).setParameter(EasyFile.DATASET_ID, id));
```

Comment:
Currently throws

    org.hibernate.HibernateException: No session currently bound to execution context

Reply:
Have a look at this code: https://github.com/DANS-KNAW/dd-manage-prestaging/blob/master/src/main/java/nl/knaw/dans/prestaging/cli/LoadFromDataverseCommand.java#L61

And also the linked Stackoverflow post. Summary: you need proxy that lets you use the `@UnitOfWork` annotation.